meta_tools/make_new_model_dir.py

```diff
@@ -1,70 +1,170 @@
-    if current_dir.match('*meta_tools'):
-        models_dir = current_dir.parent / relative_path
-    else:
-        models_dir = current_dir  /relative_path
-
-    model_dir = models_dir / model_name
-    
-    try:
-        model_dir.mkdir(parents=True, exist_ok=False)        
-        print(f"Created new model directory: {model_dir}")
-    except FileExistsError:
-        print(f"Model directory already exists: {model_dir}")
-    
-    # Create subdirectories
-    subdirs = [
-        "configs",
-        "data/raw",
-        "data/processed",
-        "data/generated",
-        "artifacts",
-        "notebooks",
-        "src/dataloaders",
-        "src/architectures",
-        "src/utils",
-        "src/visualization",
-        "src/training",
-        "src/offline_evaluation",
-        "src/forecasting",
-        "src/management"
-    ]
-    
-    for subdir in subdirs:
-        subdir_path = model_dir / subdir
-        subdir_path.mkdir(parents=True, exist_ok=True)
-        print(f"Created subdirectory: {subdir_path}")
-    
-    # Create README.md and requirements.txt
-    readme_path = model_dir / "README.md"
-    with open(readme_path, "w") as readme_file:
-        readme_file.write("# Model README\n")
-    print(f"Created README.md: {readme_path}")
-    
-    requirements_path = model_dir / "requirements.txt"
-    with open(requirements_path, "w") as requirements_file:
-        requirements_file.write("# Requirements\n")
-    print(f"Created requirements.txt: {requirements_path}")
-    
-    return model_dir
+
+    def __init__(self, model_name) -> None:
+        """
+        Initialize a ModelDirectory object with the given model name and set up paths.
+
+        Args:
+            model_name (str): The name of the model for which directories and files are to be created.
+
+        Returns:
+            None
+        """
+
+        self.expected_structure = [
+            "configs",
+            "artifacts",
+            "notebooks",
+            "reports",
+            "src/dataloaders",
+            "src/architectures",
+            "src/utils",
+            "src/training",
+            "src/offline_evaluation",
+            "src/online_evaluation",
+            "src/forecasting"
+        ]
+
+        self.current_dir = Path.cwd()
+        self.relative_path = "models"
+        self.model_name = model_name
+        if not utils_model_naming.validate_model_name(self.model_name):
+            raise ValueError(
+                "Invalid model name. Please use the format 'adjective_noun' in lowercase.")
+        # TODO: Fix this
+        # If the current directory is not the root directory, go up one level and append "models"
+        if self.current_dir.match('*meta_tools'):
+            self.models_dir = self.current_dir.parent / self.relative_path
+        else:
+            self.models_dir = self.current_dir / self.relative_path
+
+        self.model_dir = self.models_dir / self.model_name
```

Comment:
This was transplanted from the previous version of this file. I left it in because it worked. Will write up a cleaner solution for now. This issue has already been addressed in the new branch we discussed.

Reply:
Good